modules/spawner.py

```diff
@@ -15,7 +15,7 @@ def __init__ (self):
         Director.Mission.__init__(self)
 
         self.fac=[]
-        self.facnames=["confed","homeland-security","merchant","hunter","pirates","luddites","aera","rlaan","ISO","andolian","highborn","shaper","unadorned","purist","forsaken","LIHW","klkk","mechanist","shmrn","rlaan_briin"]
+        self.facnames=["ISO","LIHW","aera","andolian","bzbr","cities","confed","forsaken","highborn","homeland-security","klkk","merchant","purist","rlaan","shaper","uln","unadorned"]
```

Comment:
Two things:
1. Missing:
- hunter
- pirates
- luddites
- mechanist
- shmrn
- rlaan_briin

Are they simply not valid? Or did they just get missed/dropped?

2. I do like that they're now sorted, but a better format for this is:
```python
self.facnames = [
   "ISO",
   "LIHW",
   "aera",
   ...
   "unadorned"
]
```
This is more normative Python as it allows for easier modification of the list and better sight of the changes in a diff.

Reply:
luddites, mechanist, and shmrn definitely need to stay. They're pretty important to the storyline.

In fact, I would say all of these need to stay.

Reply:
Perhaps we need to try to locate some missing files from the old svn repos or something?

Reply:
So checking an old SVN copy (revision 13728) I see:
```
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name luddites
./win32/data/units/factions/luddites
./data/units/factions/luddites
./masters/units/factions/luddites
./mac/VegaStrike.app/Contents/Resources/units/factions/luddites
./win64/data/units/factions/luddites
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name hunter
./win32/data/communications/sounds/hunter
./data/communications/sounds/hunter
./mac/VegaStrike.app/Contents/Resources/communications/sounds/hunter
./win64/data/communications/sounds/hunter
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name pirates
./win32/data/units/factions/pirates
./win32/data/mission/cargo/pirates
./win32/data/mission/bounty/pirates
./win32/data/mission/defend/pirates
./data/units/factions/pirates
./data/mission/cargo/pirates
./data/mission/bounty/pirates
./data/mission/defend/pirates
./masters/units/factions/pirates
./translations/italiano/0.4.3/mission/cargo/pirates
./translations/italiano/0.4.3/mission/bounty/pirates
./mac/VegaStrike.app/Contents/Resources/units/factions/pirates
./mac/VegaStrike.app/Contents/Resources/mission/cargo/pirates
./mac/VegaStrike.app/Contents/Resources/mission/bounty/pirates
./mac/VegaStrike.app/Contents/Resources/mission/defend/pirates
./win64/data/units/factions/pirates
./win64/data/mission/cargo/pirates
./win64/data/mission/bounty/pirates
./win64/data/mission/defend/pirates
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name mechanist
./win32/data/communications/sounds/mechanist
./data/communications/sounds/mechanist
./mac/VegaStrike.app/Contents/Resources/communications/sounds/mechanist
./win64/data/communications/sounds/mechanist
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name shmrn
./history/species/shmrn
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$ find -type d -name rlaan_briin
bmeyer@clockwerks1:~/Devel/personal/wcu-privateer/vegastrike/vegastrike-code$
```
NOTE: There is some duplication as the SVN copy included separate copies for the various platforms and other things and this find them all.

Checking our production:
```
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name luddite
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name luddites
./units/factions/luddites
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name hunter
./communications/sounds/hunter
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name irates
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name pirates
./units/factions/pirates
./mission/cargo/pirates
./mission/bounty/pirates
./mission/defend/pirates
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name mechanist
./communications/sounds/mechanist
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name shmrn
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$ find -type d -name rlaan_briin
bmeyer@clockwerks1:~/Devel/personal/vegastrike/assets/Assets-Production$
```

So there may be some things that didn't get pulled over.

That said, it might be easier to solve this kind of thing by changing the layout. Right now I'm observing a layout where we group by ship, base, etc first and then by faction. It might be better to reverse that - group by faction, then by ship, base, etc. Even if we had a "common" that any of them could refer to as a reference. For now, just think about it as that will be a major undertaking to change.

Reply:
And the rlaan_brin as well they are a separate species from the the rlaan and have a differing attitude.

Reply:
Firstly, let's look at the files themselves:
[evert@Vorster fgnames]$ diff ISO.txt LIHW.txt
[evert@Vorster fgnames]$ diff ISO.txt andolian.txt
[evert@Vorster fgnames]$ diff ISO.txt confed.txt  
[evert@Vorster fgnames]$ diff ISO.txt forsaken.txt
[evert@Vorster fgnames]$ diff ISO.txt highborn.txt
[evert@Vorster fgnames]$ diff ISO.txt homeland-security.txt
[evert@Vorster fgnames]$ diff ISO.txt merchant.txt         
[evert@Vorster fgnames]$ diff ISO.txt purist.txt  
[evert@Vorster fgnames]$ diff ISO.txt shaper.txt
[evert@Vorster fgnames]$ diff ISO.txt unadorned.txt
These also match /universe/names.txt.

11 of the files are exactly the same... ie: every human. 
Special names here are: aera, bzbr, klkk, rlaan and uln. 

For the sake of simplicity, let's get rid of all 11 of these files, keeping only the ones that are different from universe/names.txt. I'll make sure this spawner function first looks up to see if there is special names list for the faction, and if not just uses universe/names.txt. I'll still strip out the special characters from that file. 

Of course, this means that all the factions will be put back the way they were before.